### PR TITLE
[utils] test zero tensor related APIs

### PR DESCRIPTION
**Pull request by FredHuang99**
Test all the functions that updated in colossalai.utils.memory_utils.utils

### CODE REVIEW

File: tests/test_utils/test_tensor_move.py [hunk 1]
```diff
@@ -1,34 +1,54 @@
...
 import torch
 
 from functools import partial
 import torch.multiprocessing as mp
 
+def _run_colo_tensor_mem_usage():
+    for t in [StatefulTensor(torch.randn(2, 3)), torch.randn(2, 3)]:
+        cuda_use, cpu_use = colo_tensor_mem_usage(t)
+        print("cuda_use")
+        print(cuda_use)
```
Comment (feifeibear) [inline tests/test_utils/test_tensor_move.py:18]: You did not check anything. Run the function is useless.

File: tests/test_utils/test_tensor_move.py [hunk 2]
```diff
@@ -1,34 +1,54 @@
...
+        print("cput_use")
+        print(cpu_use)
+
+def _run_colo_set_process_memory_fraction():
+    colo_set_process_memory_fraction(0.5)
+
```
Comment (feifeibear) [inline tests/test_utils/test_tensor_move.py:24]: check the available cuda memory here?

File: tests/test_utils/test_tensor_move.py [hunk 3]
```diff
@@ -1,34 +1,54 @@
...
+def _run_colo_cuda_memory_capacity():
+    frac = colo_cuda_memory_capacity
+    print("cuda memory capacity of the current cuda")
```
Comment (feifeibear) [inline tests/test_utils/test_tensor_move.py:27]: rm the print and check something useful.

File: tests/test_utils/test_tensor_move.py [hunk 4]
```diff
@@ -1,34 +1,54 @@
 import pytest
 
 from colossalai.utils.cuda import get_current_device
-from colossalai.utils.memory_utils.utils import colo_model_data_tensor_move, colo_model_data_tensor_move_inline
+from colossalai.utils.memory_utils.utils import colo_tensor_mem_usage, colo_model_data_tensor_move, colo_model_data_tensor_move_inline, colo_set_process_memory_fraction, colo_cuda_memory_capacity
```
Comment (feifeibear) [inline tests/test_utils/test_tensor_move.py:4]: You did not test this colo_model_tensor_clone()

File: tests/test_utils/test_tensor_move.py [hunk 5]
```diff
@@ -1,42 +1,83 @@
...
+def _run_colo_set_process_memory_fraction_and_colo_cuda_memory_capacity():
+    frac1 = colo_cuda_memory_capacity()
+    colo_set_process_memory_fraction(0.5)
+    frac2 = colo_cuda_memory_capacity()
+    assert frac2*2 == frac1
 
+def _run_colo_model_data_tensor_move_inline():
+    for t in [StatefulTensor(torch.randn(2,3)), torch.randn(2,3)]:
+        colo_model_data_tensor_move_inline(t, torch.device("cuda:3"))
+        assert t.device == torch.device("cuda:3")
```
Comment (feifeibear) [inline tests/test_utils/test_tensor_move.py:40]: The assert pass only when the env has 4 GPU.

see
@pytest.mark.parametrize("world_size", [1, 4])
we may run with 1 GPU.

File: tests/test_utils/test_tensor_move.py [hunk 6]
```diff
@@ -1,34 +1,54 @@
...
 
 import torch
 
 from functools import partial
 import torch.multiprocessing as mp
 
+def _run_colo_tensor_mem_usage():
+    for t in [StatefulTensor(torch.randn(2, 3)), torch.randn(2, 3)]:
+        cuda_use, cpu_use = colo_tensor_mem_usage(t)
+        print("cuda_use")
```
[the remark this answers is not in the document]
Comment (FredHuang99) [reply tests/test_utils/test_tensor_move.py]: done

File: tests/test_utils/test_tensor_move.py (hunk 1 above)
Comment (FredHuang99) [reply tests/test_utils/test_tensor_move.py]: done

File: tests/test_utils/test_tensor_move.py (hunk 2 above)
Comment (FredHuang99) [reply tests/test_utils/test_tensor_move.py]: done

File: tests/test_utils/test_tensor_move.py (hunk 3 above)
Comment (FredHuang99) [reply tests/test_utils/test_tensor_move.py]: done

File: tests/test_utils/test_tensor_move.py (hunk 4 above)
Comment (FredHuang99) [reply tests/test_utils/test_tensor_move.py]: have added it

File: tests/test_utils/test_tensor_move.py (hunk 5 above)
Comment (FredHuang99) [reply tests/test_utils/test_tensor_move.py]: change it to current_device

**Review by feifeibear — APPROVED** [review]

File: tests/test_utils/test_tensor_move.py [hunk 7]
```diff
@@ -1,42 +1,83 @@
...
+                if issubclass(type(t), StatefulTensor):
+                    assert t.payload[i][j] == p[i][j]
+                else:
+                    assert t[i][j] == p[i][j]
+
 
 
 def run_dist(rank, world_size, port):
     colossalai.launch(config={}, rank=rank, world_size=world_size, host='localhost', port=port, backend='nccl')
+    _run_colo_set_process_memory_fraction_and_colo_cuda_memory_capacity
```
Comment (feifeibear) [inline tests/test_utils/test_tensor_move.py:69]: typo

Comment (FredHuang99) [reply tests/test_utils/test_tensor_move.py]: done

**Pull request closed without merging**